Userland/Libraries/LibGfx/ImageFormats/WebPSharedLossless.h

```diff
@@ -39,6 +39,14 @@ class CanonicalCode {
     Variant<u32, Compress::CanonicalCode> m_code;
 };
 
+ALWAYS_INLINE ErrorOr<void> CanonicalCode::write_symbol(LittleEndianOutputBitStream& bit_stream, u32 symbol) const
+{
+    TRY(m_code.visit(
```

Comment:
that's concerning if `visit` is causing perf issues. It's impl is currently recursive because we have to map a run-time index to a compile-time time, but at the same time, there's only 2 types here :thinking:

Reply:
I have since upgraded from Xcode 15.2 to 15.4 and I can no longer measure an effect from these attributes here.

CxByte suggested on discord to make Variant::visit() as FLATTEN. I tried that locally, and that does still hurt perf a lot, even on 15.4.